File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/OpenApiExtraSchemaTest.java
package io.micronaut.configuration.openapi.docs;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class OpenApiExtraSchemaTest {

    @Test
    void extraSchemasAreAdded() {
        var schemas = OpenApiSpec.load().getComponents().getSchemas();

        assertTrue(schemas.containsKey("UnusedSchema"));
        assertTrue(schemas.containsKey("UnusedModel1"));
        assertTrue(schemas.containsKey("ExtraModel"));
        assertFalse(schemas.containsKey("ExcludedModel"));
        assertFalse(schemas.containsKey("ExcludedByPackage"));
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/OpenApiExtraSchemaTest.kt
package io.micronaut.configuration.openapi.docs

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class OpenApiExtraSchemaTest {

    @Test
    fun extraSchemasAreAdded() {
        val schemas = OpenApiSpec.load().components.schemas

        assertTrue(schemas.containsKey("UnusedSchema"))
        assertTrue(schemas.containsKey("UnusedModel1"))
        // KSP cannot enumerate the classes of a source package, so `packages = [...]` only applies to compiled classes
        assertFalse(schemas.containsKey("ExtraModel"))
        assertFalse(schemas.containsKey("ExcludedModel"))
        assertFalse(schemas.containsKey("ExcludedByPackage"))
    }
}
